src/main/java/com/synopsys/integration/detect/workflow/bdio/aggregation/AggregateModeSubProjectOperation.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * synopsys-detect
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.detect.workflow.bdio.aggregation;
+
+import com.synopsys.integration.bdio.graph.DependencyGraph;
+import com.synopsys.integration.detect.configuration.DetectUserFriendlyException;
+import com.synopsys.integration.detect.workflow.codelocation.DetectCodeLocation;
+
+import java.io.File;
+import java.util.List;
+
+public class AggregateModeSubProjectOperation {
```

Comment:
As I started to implement this I ran into a dilemma: If this one-liner (line 26) moves to OperationFactory, then either (1) OperationFactory creates and uses FullAggregateGraphCreator (which I think we try to avoid), or the FullAggregateGraphCreator object is injected by DetectRun. Do we want DetectRun (or maybe UtilitySingletons) knowing about a class as low-level as FullAggregateGraphCreator?

Reply:
I'm ok going either way, I could definitely see the argument of always wrapping in operation classes. There are definitely other operations we could make wrapper classes for but we currently don't and some that are simple that we do. I'm wondering when we think it makes sense and when we don't or if we just always should. 

This one seems a bit too simple to be wrapped, but maybe we should always wrap because if this got any more complicated I would want the wrap anyway. 

The operation factory already creates the FullAggregateGraphCreator that is passed to this class so all that really changes is removing the 'extra' code that is the class. Operation Factory is the one who instantiates so it does not need to be injected into the operation factory (like through detect run).

`new AggregateModeTransitiveOperation(new FullAggregateGraphCreator(new SimpleBdioFactory())).aggregateCodeLocations(directoryManager.getSourceDirectory(), detectCodeLocations);`

Versus

`new FullAggregateGraphCreator(new SimpleBdioFactory()).aggregateCodeLocations(new DependencyAggregateNodeCreator(), sourcePath, codeLocations);`

I'm totally fine keeping the wrappers but I wanted to at least discuss it!

Reply:
Yeah I don't know what I was thinking when I wrote that. The wrappers are gone.